cedar-policy-core/src/parser/err.rs

```diff
@@ -369,14 +364,10 @@ pub enum ToASTErrorKind {
     #[error(transparent)]
     #[diagnostic(transparent)]
     Unescape(#[from] UnescapeError),
-    /// Returns when a policy scope has incorrect EntityUIDs/Template Slots
-    #[error(transparent)]
-    #[diagnostic(transparent)]
-    RefCreation(#[from] RefCreationError),
-    /// Returned when an `is` appears in an invalid position in the policy scope
+    /// Returned when a policy scope has incorrect entity uids or template slots
     #[error(transparent)]
     #[diagnostic(transparent)]
-    InvalidIs(#[from] InvalidIsError),
+    RefCreation(#[from] parse_errors::RefCreation),
```

Comment:
`RefCreation` has never made much sense as an error name to me. Maybe change if you have a good idea for something better.

Reply:
The name comes from the `cst::Ref` type that describes a "reference to an entity". But we don't use this terminology anywhere else, so I agree that it's confusing as an error name. Changed in the latest commit to `WrongEntityArgument` since it (at least currently) can only be returned when a scope clause uses incorrect entity arguments (e.g., a template slot of an entity uid). I left the remaining "ref" notation in `cst_to_ast.rs` unchanged avoid exploding the size of the diff. Worth making an issue to fix later?